src/main/java/teammates/common/datatransfer/StudentAttributes.java

```diff
@@ -130,7 +129,7 @@ public boolean isRegistered() {
     }
 
     public String getRegistrationUrl() {
-        return new Url(Config.APP_URL + Const.ActionURIs.STUDENT_COURSE_JOIN_NEW)
+        return new Url(Const.ActionURIs.STUDENT_COURSE_JOIN_NEW)
```

Comment:
I still think it's unnecessary. If anything, it's because system-wide all URLs generated are relative, except for presentation purpose (in emails, in admin search, in admin home page right after account creation). No point in explaining what is commonly practiced.
Having said that, I'm now inclined to use relative URLs also in `PageData.java`. Will do that in a jiffy.